bench/clblas_benchmark.cpp

```diff
@@ -0,0 +1,353 @@
+      MemBuffer<ScalarT, CL_MEM_READ_ONLY> bufr(context, &vr, 1);
+      MemBuffer<ScalarT, CL_MEM_HOST_NO_ACCESS> scratch(context, size);
+      Event event;
+      flops = benchmark<>::measure(no_reps, size * 2, [&]() {
+        clblasXdot<ScalarT>::func(size, bufr.dev(), 0, buf1.dev(), 0, 1,
+                                  buf2.dev(), 0, 1, scratch.dev(), 1,
+                                  context._queue(), 0, NULL, &event._cl());
+        event.wait();
+        event.release();
+      });
+    }
+    return flops;
+  }
+
+  BENCHMARK_FUNCTION(iamax_bench) {
+    using ScalarT = TypeParam;
+    double flops;
+    {
+      unsigned vi;
+      MemBuffer<ScalarT, CL_MEM_WRITE_ONLY> buf1(context, size);
+      MemBuffer<unsigned, CL_MEM_READ_ONLY> buf_i(context, &vi, 1);
+      MemBuffer<ScalarT, CL_MEM_HOST_NO_ACCESS> scratch(context, 2 * size);
+      Event event;
+      flops = benchmark<>::measure(no_reps, size * 2, [&]() {
+        clblasiXamax<ScalarT>::func(size, buf_i.dev(), 0, buf1.dev(), 0, 1,
+                                    scratch.dev(), 1, context._queue(), 0, NULL,
+                                    &event._cl());
+        event.wait();
+        event.release();
+      });
+    }
+    return flops;
+  }
+
+  BENCHMARK_FUNCTION(scal2op_bench) {
+    using ScalarT = TypeParam;
+    double flops;
+    {
+      ScalarT alpha(2.4367453465);
+      MemBuffer<ScalarT> buf1(context, size);
+      MemBuffer<ScalarT> buf2(context, size);
+      Event event1, event2;
+      flops = benchmark<>::measure(no_reps, size * 2, [&]() {
+        clblasXscal<ScalarT>::func(size, alpha, buf1.dev(), 0, 1, 1,
+                                   context._queue(), 0, NULL, &event1._cl());
+        clblasXscal<ScalarT>::func(size, alpha, buf2.dev(), 0, 1, 1,
+                                   context._queue(), 0, NULL, &event2._cl());
+        Event::wait({event1, event2});
+        Event::release({event1, event2});
+      });
+    }
+    return flops;
+  }
+
+  BENCHMARK_FUNCTION(scal3op_bench) {
+    using ScalarT = TypeParam;
+    double flops;
+    {
+      ScalarT alpha(2.4367453465);
+      MemBuffer<ScalarT> buf1(context, size);
+      MemBuffer<ScalarT> buf2(context, size);
+      MemBuffer<ScalarT> buf3(context, size);
+      Event event1, event2, event3;
+      flops = benchmark<>::measure(no_reps, size * 3, [&]() {
+        clblasXscal<ScalarT>::func(size, alpha, buf1.dev(), 0, 1, 1,
+                                   context._queue(), 0, NULL, &event1._cl());
+        clblasXscal<ScalarT>::func(size, alpha, buf2.dev(), 0, 1, 1,
+                                   context._queue(), 0, NULL, &event2._cl());
+        clblasXscal<ScalarT>::func(size, alpha, buf3.dev(), 0, 1, 1,
+                                   context._queue(), 0, NULL, &event3._cl());
+        Event::wait({event1, event2, event3});
+        Event::release({event1, event2, event3});
+      });
+    }
+    return flops;
+  }
+
+  BENCHMARK_FUNCTION(axpy3op_bench) {
```

Comment:
It would have been really interesting to talk about the operations before to be defined.
Have you thought to define the same operations for MAGMA?